google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/JsonStreamWriter.java

```diff
@@ -0,0 +1,377 @@
+
+    StreamWriter.Builder streamWriterBuilder;
+    if (this.client == null) {
+      streamWriterBuilder = StreamWriter.newBuilder(builder.streamName);
+    } else {
+      streamWriterBuilder = StreamWriter.newBuilder(builder.streamName, builder.client);
+    }
+    setStreamWriterSettings(
+        streamWriterBuilder,
+        builder.channelProvider,
+        builder.credentialsProvider,
+        builder.batchingSettings,
+        builder.retrySettings,
+        builder.executorProvider,
+        builder.endpoint,
+        builder.onSchemaUpdateRunnable);
+    this.streamWriter = streamWriterBuilder.build();
+  }
+  /**
+   * Writes a JSONArray that contains JSONObjects to the BigQuery table by first converting the JSON
+   * data to protobuf messages, then using StreamWriter's append() to write the data. If there is a
+   * schema update, the OnSchemaUpdateRunnable will be used to determine what actions to perform.
+   *
+   * @param jsonArr The JSON array that contains JSONObjects to be written
+   * @param allowUnknownFields if true, json data can have fields unknown to the BigQuery table.
+   * @return ApiFuture<AppendRowsResponse> returns an AppendRowsResponse message wrapped in an
+   *     ApiFuture
+   */
+  public synchronized ApiFuture<AppendRowsResponse> append(
+      JSONArray jsonArr, boolean allowUnknownFields) {
+    return append(jsonArr, -1, allowUnknownFields);
+  }
+
+  /**
+   * Writes a JSONArray that contains JSONObjects to the BigQuery table by first converting the JSON
+   * data to protobuf messages, then using StreamWriter's append() to write the data. If there is a
+   * schema update, the OnSchemaUpdateRunnable will be used to determine what actions to perform.
+   *
+   * @param jsonArr The JSON array that contains JSONObjects to be written
+   * @param offset Offset for deduplication
+   * @param allowUnknownFields if true, json data can have fields unknown to the BigQuery table.
+   * @return ApiFuture<AppendRowsResponse> returns an AppendRowsResponse message wrapped in an
+   *     ApiFuture
+   */
+  public synchronized ApiFuture<AppendRowsResponse> append(
+      JSONArray jsonArr, long offset, boolean allowUnknownFields) {
+    ProtoRows.Builder rowsBuilder = ProtoRows.newBuilder();
+    // Any error in convertJsonToProtoMessage will throw an
+    // IllegalArgumentException/IllegalStateException/NullPointerException and will halt processing
+    // of JSON data.
+    for (int i = 0; i < jsonArr.length(); i++) {
+      JSONObject json = jsonArr.getJSONObject(i);
+      Message protoMessage =
+          JsonToProtoMessage.convertJsonToProtoMessage(this.descriptor, json, allowUnknownFields);
+      rowsBuilder.addSerializedRows(protoMessage.toByteString());
+    }
+    AppendRowsRequest.ProtoData.Builder data = AppendRowsRequest.ProtoData.newBuilder();
+    data.setWriterSchema(ProtoSchemaConverter.convert(this.descriptor));
```

Comment:
Don't append writer schema to the request for every request. It only needs to be set for the first request. You can see how DirectWriter handles this.

Reply:
Do we also need to set AppendRowsRequest.write_stream?

Reply:
Resolved: StreamWriter will remove the schemas.